mitosheet/src/DataFrameMimeRenderer.tsx

```diff
@@ -0,0 +1,132 @@
+
+            // Clean up the script after execution
+            return () => {
+                document.body.removeChild(script);
+            };
+        }
+    }, [props.jsCode]);
+
+    return (
+        <div
+            dangerouslySetInnerHTML={{ __html: props.htmlContent }}
+        ></div>
+    );
+};
+
+export class DataFrameMimeRenderer extends Widget implements IRenderMime.IRenderer {
+    private _app: JupyterFrontEnd;
+    private _notebookTracker: INotebookTracker;
+
+    constructor(app: JupyterFrontEnd, options: IRenderMime.IRendererOptions, notebookTracker: INotebookTracker) {
+        super();
+        this._app = app;
+        this.addClass(CLASS_NAME);
+        this._notebookTracker = notebookTracker;
+    }
+
+    async renderModel(model: IRenderMime.IMimeModel): Promise<void> {
+        this.node.innerHTML = '';  // Clear previous content
+        const originalRawData = model.data['text/html']?.toString();
+        const isDataframeOutput = originalRawData?.includes('class="dataframe"');
+
+
+        const notebook = this._notebookTracker.currentWidget?.content;
+        const cells = notebook?.model?.cells;
+        const activeCellIndex = notebook?.activeCellIndex
+
+        let dataframeVariableName = undefined;
+        if (activeCellIndex) {
+            const previousCell = getCellAtIndex(cells, activeCellIndex - 1)
+            dataframeVariableName = getLastNonEmptyLine(getCellText(previousCell))
+        }
+
+
+        if (isDataframeOutput) {
+            console.log('Dataframe detected!!!!');
+
+            // Define the Python code to run
+            const pythonCode = `mitosheet.sheet(${dataframeVariableName || ''} )`;
+
+            try {
+                const notebookPanel = this._notebookTracker.currentWidget;
+                const kernel = notebookPanel?.context.sessionContext.session?.kernel;
+
+
+                if (!kernel) {
+                    console.error('No active kernel found.');
+                    return;
+                }
+
+                console.log("GOT A KERNEL")
+
+                // Execute the Python code
+                const future = kernel.requestExecute({ code: pythonCode });
+
+                future.onIOPub = (msg: any) => {
+                    if (msg.header.msg_type === 'execute_result' || msg.header.msg_type === 'display_data') {
+                        const htmlOutput = msg.content.data['text/html'];
+                        if (htmlOutput) {
+                            console.log('Received HTML output:', htmlOutput);
+                            // Extract the javascript code so we can execute it when we render the output.
+                            // Remember, that the javascript code is actually what creates the sheet interface!
+                            const scriptMatch = htmlOutput.match(/<script[^>]*>([\s\S]*?)<\/script>/);
+                            const jsCode = scriptMatch ? scriptMatch[1] : '';
```

Comment:
Explore what the other elements in the scriptMatch are. Leave a comment here to explain